From dab7c001aaf3f5863873f51a22f0c00b993f52ab Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Wed, 20 Feb 2019 14:47:04 +0530
Subject: [PATCH] tls modules: Export load/unload/query symbols on MSVC

These must be explicitly exported or the DLLs outputted won't be
loadable

'g_io_gnutls_load': The specified procedure could not be found.
Failed to load module: c:\path\to\lib\gio\modules\giognutls.dll
---
 tls/gnutls/gnutls-module.c   | 6 +++---
 tls/openssl/openssl-module.c | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/tls/gnutls/gnutls-module.c b/tls/gnutls/gnutls-module.c
index 8983bd7..8dd09ca 100644
--- a/tls/gnutls/gnutls-module.c
+++ b/tls/gnutls/gnutls-module.c
@@ -30,7 +30,7 @@
 #include "gtlsbackend-gnutls-pkcs11.h"
 
 
-void
+G_MODULE_EXPORT void
 g_io_gnutls_load (GIOModule *module)
 {
   gchar *locale_dir;
@@ -53,12 +53,12 @@ g_io_gnutls_load (GIOModule *module)
   g_free (locale_dir);
 }
 
-void
+G_MODULE_EXPORT void
 g_io_gnutls_unload (GIOModule *module)
 {
 }
 
-gchar **
+G_MODULE_EXPORT gchar **
 g_io_gnutls_query (void)
 {
   gchar *eps[] = {
-- 
2.20.1

